src/util/http/websocket/WebSocketManager.cpp

```diff
@@ -146,11 +71,13 @@ net::awaitable<void> WebSocketManager::connectionLifecycle(
 
     auto strand = net::make_strand(ws.get_executor());
 
-    // experimental operators
-    co_await (
-        net::co_spawn(strand, waitForServerEvents(ws, queryId),
-                      net::use_awaitable) &&
-        net::co_spawn(strand, handleClientCommands(ws), net::use_awaitable));
+    co_await net::dispatch(strand, net::use_awaitable);
```

Comment:
I currently don't like that the `closing of the websocket` is the implicit signal for `normal termination`.
I think the semantics that we want is
`wait until one of them either finishes or fails, and then close the websocket if still possible. But it might be, that implementing this would also not lead to nicer code.